server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/GetVacationResponseMethod.java

```diff
@@ -64,27 +64,27 @@ public GetVacationResponseMethod(VacationRepository vacationRepository, ZonedDat
     }
 
     @Override
-    public Stream<JmapResponse> process(JmapRequest request, MethodCallId methodCallId, MailboxSession mailboxSession) {
+    public Flux<JmapResponse> process(JmapRequest request, MethodCallId methodCallId, MailboxSession mailboxSession) {
         Preconditions.checkNotNull(request);
         Preconditions.checkNotNull(methodCallId);
         Preconditions.checkNotNull(mailboxSession);
         Preconditions.checkArgument(request instanceof GetVacationRequest);
 
+        return Flux.from(metricFactory.runPublishingTimerMetricLogP99(JMAP_PREFIX + METHOD_NAME.getName(),
+            process(mailboxSession)
+                .flatMapMany(response -> Flux.just(JmapResponse.builder()
+                    .methodCallId(methodCallId)
+                    .responseName(RESPONSE_NAME)
+                    .response(response)
+                    .build()))));
+    }
 
-        return MDCBuilder.create()
```

Comment:
are we loosing MDC?

Reply:
Yes, MDC don't seem to be easily compatible with reactor...

We may need to rework our structured logging simplistic solution.

Reply:
read https://projectreactor.io/docs/core/release/reference/index.html#faq.mdc